src/commands/lambda/flare.ts

```diff
@@ -1,14 +1,194 @@
+import * as fs from 'fs'
+import * as path from 'path'
+import util from 'util'
+
+import {LambdaClient, LambdaClientConfig} from '@aws-sdk/client-lambda'
+import {AwsCredentialIdentity} from '@aws-sdk/types'
+import axios from 'axios'
 import {Command} from 'clipanion'
+import FormData from 'form-data'
+import JSZip from 'jszip'
 
-import {renderLambdaFlareHeader} from './renderers/flare-renderer'
+import {API_KEY_ENV_VAR, AWS_DEFAULT_REGION_ENV_VAR, CI_API_KEY_ENV_VAR} from './constants'
+import {getAWSCredentials, getLambdaFunctionConfig} from './functions/commons'
+import {requestAWSCredentials} from './prompt'
+import {renderError, renderLambdaFlareHeader, renderSoftWarning} from './renderers/flare-renderer'
 
 export class LambdaFlareCommand extends Command {
   private isDryRun = false
+  private functionName?: string
+  private region?: string
+  private apiKey?: string
+  private caseId?: string
+  private email?: string
+  private credentials?: AwsCredentialIdentity
 
+  /**
+   * @returns 0 if the command ran successfully, 1 otherwise.
+   */
   public async execute() {
     this.context.stdout.write(renderLambdaFlareHeader(this.isDryRun))
+
+    // Validate function name
+    let errorFound = false
+    if (this.functionName === undefined) {
+      this.context.stderr.write(renderError('No function name specified. [-f,--function]'))
+      errorFound = true
+    }
+
+    // Validate region
+    if (
+      this.region === undefined &&
+      this.functionName !== undefined &&
+      this.functionName.startsWith('arn:aws:lambda')
+    ) {
+      this.region = this.functionName.split(':')[3]
+    }
+    this.region = this.region ?? process.env[AWS_DEFAULT_REGION_ENV_VAR]
```

Comment:
```suggestion
    if (
      this.region === undefined &&
      this.functionName !== undefined &&
      this.functionName.startsWith('arn:aws:lambda')
    ) {
      this.region = this.functionName.split(':')[3]
    }
    this.region = this.region ?? process.env[AWS_DEFAULT_REGION_ENV_VAR]
```

Also, there is already a function that gives you the region given an ARN.